src/components/App/App.js

```diff
@@ -180,14 +165,23 @@ function App() {
             path="/farmers/:id"
             isAllowed={isLoggedIn()}
             redirectTo="/login"
-            render={props => (
-              <DisplayFarmer
-                {...props}
-                farmers={data.farmers}
-                getFarmer={getFarmer}
-                needsUpdate={setNeedsUpdate}
-              />
-            )}
+            render={props => {
+              const id = props.match.params.id;
+              let selectedFarmer;
+              if (data.farmers) {
+                selectedFarmer = getFarmer(id);
+                if (!selectedFarmer) selectedFarmer = null;
+              }
+
+              return (
+                <DisplayFarmer
+                  {...props}
+                  farmer={selectedFarmer}
+                  getFarmer={getFarmer}
```

Comment:
As far as I can tell, this prop is not needed in DisplayFarmer

Reply:
Correct, I moved the function call into to the router and I forgot to remove the prop